src/paradigma/tremor/feature_extraction.py

```diff
@@ -0,0 +1,431 @@
+        l_values_total.append(l_values)
+        l_freqs_total.append(l_freqs)
+
+    return l_freqs_total, l_values_total
+
+def compute_spectrogram(
+        values: list,
+        window_type: str = 'hann',
+        sampling_frequency: int = 100,
+        segment_length_s: float = 2,
+        overlap: float = 0.8,
+    )-> tuple:
+    
+    """Compute the spectrogram (using short time fourier transform) of the gyroscope signal
+    
+    Parameters
+    ----------
+    values: List
+        The values of the signal (e.g., gyroscope data) of a single window.
+    window_type: str
+        The type of window to be used for the spectrogram (default: 'hann')
+    sampling_frequency: int
+        The sampling frequency of the signal (default: 100)
+    segment_length_s: float
+        The length of each segment in seconds (default: 2)
+    overlap: float
+        The overlap between segments as fraction (default: 0.8)
+        
+    Returns
+    -------
+    tuple
+        Lists of spectrogram values
+    """
+
+    segment_length_n = sampling_frequency*segment_length_s
+    overlap_n = segment_length_n*overlap
+    window = signal.get_window(window_type,segment_length_n)
+
+    f, t, S1 = signal.stft(values, fs=sampling_frequency, window=window, nperseg=segment_length_n, 
+                           noverlap=overlap_n,boundary=None)
+    S = np.abs(S1)*sampling_frequency 
+    return S
+
+def signal_to_spectrogram(
+        sensor_col: pd.Series,
+        window_type: str = 'hann',
+        sampling_frequency: int = 100,
+        segment_length_s: float = 2,
+        overlap: float = 0.8,
+    ) -> list:
+    
+    """Compute the spectrogram (using short time fourier transform) of a signal per window.
+
+    Parameters
+    ----------
+    sensor_col: pd.Series
+        The sensor column to be transformed (e.g. x-axis of gyroscope)
+    window_type: str
+        The type of window to be used for the spectrogram (default: 'hann')
+    sampling_frequency: int
+        The sampling frequency of the signal (default: 100)
+    segment_length_s: float
+        The length of each segment in seconds (default: 2)
+    overlap: float
+        The overlap between segments as fraction (default: 0.8)
+    
+    Returns
+    -------
+    tuple
+        Lists of spectrogram values which can be concatenated as column to the dataframe
+    """
+    spectrogram = []
+    for row in sensor_col:
+        spectrogram_values = compute_spectrogram(
+            values=row,
+            window_type=window_type,
+            sampling_frequency=sampling_frequency,
+            segment_length_s = segment_length_s,
+            overlap = overlap
+            )
+        spectrogram.append(spectrogram_values)
+
+    return spectrogram
+
+def melscale(x):
+    "Maps values of x to the melscale"
+    y = 64.875 * np.log10(1 + x/17.5)
+    return y
+
+def inverse_melscale(x):
+    "Inverse of the melscale"
+    y = 17.5 * (10**(x/64.875) - 1)
```

Comment:
```suggestion
    return 17.5 * (10**(x/64.875) - 1)
```

Reply:
Voor leesbaarheid is het handig om spaties tussen operators (=, +, -, /, *, etc.) te plaatsen.